pkg/internal/apis/config/validate.go

```diff
@@ -153,7 +152,11 @@ func validateSubnets(subnetStr string, dualstack bool) error {
 		subnets = append(subnets, cidr)
 	}
 
+	dualstack := ipFamily == DualStackFamily
 	switch {
+	// if no subnets are defined
+	case len(subnets) == 0:
+		allErrs = append(allErrs, errors.New("no subnets defined"))
```

Comment:
IIRC defaulting is before validation, but in case of having zero here, is this an error or is a "no need validation"  @BenTheElder ?

Reply:
we only added this case because https://github.com/kubernetes-sigs/kind/pull/2473/files#r714772513 :-)

IIRC defaulting does come first so maybe we should be throwing a "this can't happen" type error (maybe even panic?) /shrug

Reply:
We can have an error and add a test for protecting of future changes that may remove the defaulting, if we panic and we don't test it we never now until someone hits the panic

@fedepaol can you add a test that it fails without subnets with a comment explaining that this should never happen for future us?

Reply:
Will do